File: 프로그래머스/2/87946. 피로도/피로도.cs
using System;

public class Solution {
    public bool[] visited;
    public int answer = 0;
    public int solution(int k, int[,] dungeons) {
        visited = new bool[dungeons.GetLength(0)];
        
        BackTracking(0, k, dungeons);
        return answer;
    }
    public void BackTracking(int depth, int k, int[,] dungeons)
    {
        for(int i = 0; i < dungeons.GetLength(0); i++)
        {
            if(!visited[i] && dungeons[i, 0] <= k)
            {
                visited[i] = true;
                BackTracking(depth + 1, k - dungeons[i, 1], dungeons);
                visited[i] = false;
            }
        }
        answer = Math.Max(depth, answer);
    }
}

File: 프로그래머스/2/87946. 피로도/피로도.cpp
#include <string>
#include <vector>

using namespace std;

int answer = 0;
bool visited[9];

void dfs(int cur, int cnt, vector<vector<int>> &dungeons)
{
    answer = max(answer, cnt);
    
    for(int i = 0; i < dungeons.size(); i++)
    {
        if(!visited[i] && cur >= dungeons[i][0])
        {
            visited[i] = true;
            dfs(cur - dungeons[i][1], cnt + 1, dungeons);
            visited[i] = false;
        }
    }
}
int solution(int k, vector<vector<int>> dungeons) {
    dfs(k, 0, dungeons);
    return answer;
}